### PR TITLE
fix(chain): stop banning peers incorrectly

### PR DESCRIPTION
**Pull request by bowenwang1996**
Currently we do not reset `stalling_ts` between header syncs. This means that once `stalling_ts` is set, it is never unset unless we restart the node. Therefore, when there is an ongoing block sync and we sync headers at the same time, there is a chance that block sync is not due before header sync stalls, which leads to a peer being banned. This PR fixes it by introducing `prev_highest_height` to `prev_header_sync` to allow resetting `stalling_ts` when a round of header sync is done.

Test plan
----------
* python test `sync_ban.py` and `sync_should_ban.py` pass.

### CODE REVIEW

**Comment by SkidanovAlex** [comment]
Haven't had a chance to dig into the logic yet, but the tests appear to only have ~10 lines that they do not share. Why not make it a single test with a command line argument?

Also, add to nightly.txt

File: pytest/tests/sanity/sync_ban.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    status = nodes[0].get_status()
+    cur_height = status['sync_info']['latest_block_height']
+
+    status1 = nodes[1].get_status()
+    node1_height = status1['sync_info']['latest_block_height']
+    if abs(node1_height - cur_height) < 5 and status1['sync_info']['syncing'] is False:
+        break
+    time.sleep(2)
+
```
Comment (SkidanovAlex) [inline pytest/tests/sanity/sync_ban.py:87]: Check that neither log has "ban a fraudlent peer"?

File: pytest/tests/sanity/sync_ban.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import sys, time, functools, asyncio
```
Comment (SkidanovAlex) [inline pytest/tests/sanity/sync_ban.py:1]: Describe what the test does in a comment at the beginning. It's very hard later (or even now) to read tests without understanding what is the intended behavior and what exactly they test.

File: chain/client/src/sync.rs [hunk 3]
```diff
@@ -65,7 +65,7 @@ pub fn highest_height_peer(highest_height_peers: &Vec<FullPeerInfo>) -> Option<F
 pub struct HeaderSync {
     network_adapter: Arc<dyn NetworkAdapter>,
     history_locator: Vec<(BlockHeight, CryptoHash)>,
-    prev_header_sync: (DateTime<Utc>, BlockHeight, BlockHeight),
+    prev_header_sync: (DateTime<Utc>, BlockHeight, BlockHeight, BlockHeight),
```
Comment (frol) [inline chain/client/src/sync.rs:68]: Please, create a struct with named fields

**Pull request merged**